x/eibc/client/cli/query_command_orders.go

```diff
@@ -1,30 +1,63 @@
 package cli
 
 import (
-	"context"
+	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/cosmos/cosmos-sdk/client"
 	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/dymensionxyz/dymension/v3/x/eibc/types"
 	"github.com/spf13/cobra"
+
+	commontypes "github.com/dymensionxyz/dymension/v3/x/common/types"
+	"github.com/dymensionxyz/dymension/v3/x/eibc/types"
 )
 
 func CmdListDemandOrdersByStatus() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "list-demand-orders [status]",
+		Use:   "list-demand-orders status [rollapp_id] [type] [limit]",
 		Short: "List all demand orders with a specific status",
-		Long:  `Query demand orders filtered by status. Examples of status include "pending", "finalized", and "reverted".`,
-		Args:  cobra.ExactArgs(1),
+		Long: `Query demand orders filtered by status. Examples of status include "pending", "finalized", and "reverted".
+				Optional arguments include rollapp_id, type (recv, timeout, ack), and limit.`,
+		Args: cobra.MinimumNArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx := client.GetClientContextFromCmd(cmd)
-			status := args[0]
+			status, ok := commontypes.Status_value[strings.ToUpper(args[0])]
+			if !ok {
+				return fmt.Errorf("invalid status: %s", args[0])
+			}
+			request := &types.QueryDemandOrdersByStatusRequest{
+				Status: commontypes.Status(status),
+				Type:   commontypes.Type_UNDEFINED, // default to undefined, as '0' is a valid type
+			}
 
+			if len(args) > 1 {
+				request.RollappId = args[1]
+			}
+			if len(args) > 2 {
+				packetType := strings.ToUpper(args[2])
+				if !strings.HasPrefix(packetType, "ON_") {
+					packetType = "ON_" + packetType
+				}
+				ptype, ok := commontypes.Type_value[packetType]
+				if !ok {
+					return fmt.Errorf("invalid packet type: %s", args[2])
+				}
+				request.Type = commontypes.Type(ptype)
+			}
+			if len(args) > 3 {
+				limit, err := strconv.ParseInt(args[3], 10, 32)
+				if err != nil {
+					return fmt.Errorf("limit must be an integer: %s", args[3])
+				}
+				request.Limit = int32(limit)
```

Comment:
Any reason why we use limit instead of Pagination in cosmos-sdk?

Reply:
well, the intent of this added filtering is just for debugging, it's not supposed to be for end users